src/Cake.Common/Tools/Chocolatey/ChocolateyToolResolver.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Linq;
+using Cake.Core;
+using Cake.Core.IO;
+
+namespace Cake.Common.Tools.Chocolatey
+{
+    /// <summary>
+    /// Contains Chocolatey path resolver functionality
+    /// </summary>
+    public sealed class ChocolateyToolResolver : IChocolateyToolResolver
+    {
+        private readonly IFileSystem _fileSystem;
+        private readonly ICakeEnvironment _environment;
+        private FilePath _chocolateyPath;
```

Comment:
Yip, that is the CA error that I mentioned above, will fix that up.

For this part, I was comparing between SignToolResolver, and NugetToolResolver, and both do things in a slightly different way, and I ended up with an amalgamation of them both :smile: